bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudDeviceHandler.java

```diff
@@ -91,35 +96,37 @@ public void deviceInformationUpdate(List<ICloudDeviceInformation> deviceInformat
 
     @Override
     public void initialize() {
-        Bridge bridge = getBridge();
-        Object bridgeStatus = (bridge == null) ? null : bridge.getStatus();
-        logger.debug("initializeThing thing [{}]; bridge status: [{}]", getThing().getUID(), bridgeStatus);
-
         ICloudDeviceThingConfiguration configuration = getConfigAs(ICloudDeviceThingConfiguration.class);
         this.deviceId = configuration.deviceId;
 
-        ICloudAccountBridgeHandler handler = getIcloudAccount();
-        if (handler != null) {
-            refreshData();
-        } else {
-            updateStatus(OFFLINE, BRIDGE_UNINITIALIZED, "Bridge not found");
-        }
-    }
+        Bridge bridge = getBridge();
+        Object bridgeStatus = (bridge == null) ? null : bridge.getStatus();
+        this.logger.debug("initializeThing thing [{}]; bridge status: [{}]", getThing().getUID(), bridgeStatus);
 
-    private void refreshData() {
-        ICloudAccountBridgeHandler bridge = getIcloudAccount();
         if (bridge != null) {
-            bridge.refreshData();
+            ((ICloudAccountBridgeHandler) bridge.getHandler()).registerListener(this);
+        }
+
+        if (bridge == null || bridge.getStatus() == ThingStatus.UNINITIALIZED) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.BRIDGE_UNINITIALIZED);
```

Comment:
If bridge is null, this would be a configuration error, i.e. OFFLINE + CONFIGURATION_ERROR. If bridge is uninitialized it will not try to initialize child things, so this doesn't need to be handled explicitly.

Reply:
Did it get it right, your suggestion is to remove this if clause, because the potential situation is handled by the framework?

Reply:
Did change this according to my understanding, maybe double check if this was your intended change.